core/src/main/java/com/amplifyframework/core/plugin/Plugin.java

```diff
@@ -16,36 +16,67 @@
 package com.amplifyframework.core.plugin;
 
 import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.WorkerThread;
 
 import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
 import com.amplifyframework.core.category.CategoryTypeable;
 
 import org.json.JSONObject;
 
 /**
- * Interface that defines the contract that every plugin
- * in Amplify System will adhere to.
- * @param <E> The class type of the escape hatch used to circumvent the generic plugin APIs
+ * A plugin is an implementation of a category's behavior. You can implement
```

Comment:
> A plugin is an implementation of a category's behavior

Well technically `Plugin` by itself is just an interface... It promises that it's configurable & that it always belongs to a certain category. But to be honest, I'm not sure how better this structure can be documented so I think this is fine

Reply:
Eh, yea - just now I tried to make it more technically accurate, I ended up with things like "An implementation of a plugin is an implementation of ..." which is confusing in-and-of-itself.